assignment/tool/src/java/org/sakaiproject/assignment/entityproviders/AssignmentEntityProvider.java

```diff
@@ -1525,6 +1603,42 @@ public SimpleAssignment(Assignment a) {
             this.estimate = a.getEstimate();
 
             this.allowPeerAssessment = a.getAllowPeerAssessment();
+
+            Site site = null;
+            try {
+                site = siteService.getSite(a.getContext());
+            } catch (IdUnusedException e) {
+                throw new EntityNotFoundException("No site found", a.getContext(), e);
+            }
```

Comment:
This would be ideal if it were at the top of the method to avoid needless work if the site doesn't exist

Reply:
done